llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -2229,6 +2229,14 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
         return BitOp;
     }
 
+    // fshl(X, 0, Y) --> shl(X, and(Y, BitWidth - 1)) if bitwidth is a
+    // power-of-2
+    if (isPowerOf2_32(BitWidth) && match(Op1, m_ZeroInt())) {
```

Comment:
Aren't we missing a check that this is fshl rather than fshr? Can you please add a test using fshr?

Reply:
But we already have a check that it's `fshl`

https://github.com/llvm/llvm-project/blob/212cba0ef37dd3b2a253c063240370de42fc67c1/llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp#L2205-L2208

Reply:
This is only for constant shift amount.

Reply:
Yes, you are right, i will add a check and update tests

Reply:
Done